distribution/errors.go

```diff
@@ -213,6 +213,12 @@ func (e reservedNameError) Error() string {
 
 func (e reservedNameError) Forbidden() {}
 
+const deprecatedSchema1 = "[DEPRECATION NOTICE] Docker Image Format v1 and Docker Image manifest version 2, schema 1 support is disabled by default and will be removed in an upcoming release. More information at https://docs.docker.com/go/deprecated-image-specs/"
+
+type invalidArgumentErr struct{ error }
+
+func (invalidArgumentErr) InvalidParameter() {}
+
 func DeprecatedSchema1ImageMessage(ref reference.Named) string {
-	return fmt.Sprintf("[DEPRECATION NOTICE] Docker Image Format v1, and Docker Image manifest version 2, schema 1 support will be removed in an upcoming release. Suggest the author of %s to upgrade the image to the OCI Format, or Docker Image manifest v2, schema 2. More information at https://docs.docker.com/go/deprecated-image-specs/", ref)
+	return fmt.Sprintf("[DEPRECATION NOTICE] Docker Image Format v1 and Docker Image manifest version 2, schema 1 support is disabled by default and will be removed in an upcoming release. Suggest the author of %s to upgrade the image to the OCI Format or Docker Image manifest v2, schema 2. More information at https://docs.docker.com/go/deprecated-image-specs/", ref)
```

Comment:
The env var should be printed here as a hint?

Reply:
I was a bit in doubt on adding even more text to the already length error, and considered instead to include that information in the deprecation docs; we can point the "/go/" redirect to the appropriate location (which can outline the details, as well as the temporary escape hatch).

It might be good for those that need it to understand the context, and be aware that they really should not depend on it.


Happy to hear you (and other's) thoughts on that though!

Reply:
Yeah I think it's better to put more info on the deprecation page. It's also an opportunity to provide more information and mention that this env var needs to be put on the daemon side (not just `DOCKER_ENABLE_DEPRECATED_PULL_SCHEMA_1_IMAGE=1 docker pull`)